### PR TITLE
feat: analyzes embedding

### PR DESCRIPTION
**Pull request by tyyzqmf**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: frontend/src/pages/analytics/comps/ExploreEmbedFrame.tsx [hunk 1]
```diff
@@ -10,67 +10,61 @@
...
+      case 'console':
+        await embeddingContext.embedConsole(
+          {
+            url: embedUrl,
+            container: `#EmbedId`,
+            resizeHeightOnSizeChangedEvent: true,
+            height: '1000',
+          },
+          {
+            locale: 'en-US',
```
Comment (zxkane) [inline frontend/src/pages/analytics/comps/ExploreEmbedFrame.tsx:52]: use the solution's language instead?

**Review by zxkane — CHANGES_REQUESTED** [review]

File: frontend/src/pages/analytics/analyzes/AnalyticsAnalyzes.tsx [hunk 2]
```diff
@@ -25,22 +25,46 @@
...
+        setDashboardEmbedUrl(data.EmbedUrl);
+      }
+    } catch (error) {
+      setLoadingData(false);
+    }
+    setLoadingData(false);
+  };
+
+  useEffect(() => {
+    getAnalyzes();
```
Comment (chenhaiyun) [inline frontend/src/pages/analytics/analyzes/AnalyticsAnalyzes.tsx:66]: add `if (projectId)`

File: src/control-plane/backend/lambda/api/service/project.ts [hunk 3]
```diff
@@ -228,6 +228,30 @@ export class ProjectServ {
...
+  public async getAnalyzes(req: any, res: any, next: any) {
+    try {
+      const { projectId } = req.params;
+      const { allowedDomain } = req.query;
+      const latestPipelines = await store.listPipeline(projectId, 'latest', 'asc');
+      if (latestPipelines.length === 0) {
+        return res.status(404).json(new ApiFail('The latest pipeline not found.'));
+      }
+      const latestPipeline = latestPipelines[0];
+      console.log(latestPipeline);
```
Comment (chenhaiyun) [inline src/control-plane/backend/lambda/api/service/project.ts:240]: remove this

File: frontend/src/pages/analytics/comps/ExploreEmbedFrame.tsx (hunk 1 above)
Comment (tyyzqmf) [reply frontend/src/pages/analytics/comps/ExploreEmbedFrame.tsx:52]: solution's language save in localStorage that is 'en', Quicksight's locale is 'en-US'.

Comment (chenhaiyun) [reply frontend/src/pages/analytics/comps/ExploreEmbedFrame.tsx:52]: > @chenhaiyun, could we specify en-US and then fallback to en in the front end?

Do you mean we save the solutions's default language to localStorage from `en` to  `en-US`?

Comment (zxkane) [reply frontend/src/pages/analytics/comps/ExploreEmbedFrame.tsx:52]: Yes. Then front end falls back to use language `en` if `en-US` is missing. And the same for using `zh` if `zh-CN` is missing.

Comment (chenhaiyun) [reply frontend/src/pages/analytics/comps/ExploreEmbedFrame.tsx]: Change the i18n folder name and set default to `zh-CN` and `en-US`

**Review by chenhaiyun — APPROVED** [review]

File: frontend/src/pages/analytics/comps/ExploreEmbedFrame.tsx (hunk 1 above)
Comment (tyyzqmf) [reply frontend/src/pages/analytics/comps/ExploreEmbedFrame.tsx]: [explore api](https://github.com/awslabs/clickstream-analytics-on-aws/blob/main/src/control-plane/backend/lambda/api/test/api/reporting.test.ts#L232) use this language save in localStorage.

File: frontend/src/pages/analytics/analytics-utils.ts [hunk 4]
```diff
@@ -445,5 +445,5 @@ export const getWarmUpParameters = (
 };
 
 export const getLngFromLocalStorage = () => {
-  return localStorage.getItem('i18nextLng') ?? 'en';
+  return localStorage.getItem('i18nextLng') ?? DEFAULT_EN_LANG;
```
[the remark this answers is not in the document]
Comment (llmin) [reply frontend/src/pages/analytics/analytics-utils.ts:448]: changed to standard locale.

**Review by zxkane — APPROVED** [review]

**Pull request merged**